src/FlagHandler.cc

```diff
@@ -44,6 +44,24 @@ FlagHandler FlagHandler::addFields(
     return r;
 }
 
+FlagHandler FlagHandler::addFields(
+    afw::table::Schema & schema,
+    std::string const & prefix,
+    std::vector<FlagDefinition> const * flagDefs
+) {
+    FlagHandler r;
+    r._vector.reserve(flagDefs->size());
+    for (unsigned int i = 0; i < flagDefs->size(); i++) {
+        r._vector.push_back(
+            std::make_pair(
+                flagDefs->at(i),
+                schema.addField<afw::table::Flag>(schema.join(prefix, flagDefs->at(i).name), flagDefs->at(i).doc)
+            )
+        );
+    }
+    return r;
```

Comment:
Since we are now using c++ 11, please switch this to a range style for loop. It will make the code more readable, and remove all the extra dereferencing ->